pkg/util/ciutil/vars.go

```diff
@@ -34,36 +34,90 @@ type Vars struct {
 	BranchName string
 	// CommitMessage is the full message of the Git commit being built.
 	CommitMessage string
+	// PRNumber is the pull-request ID/number in the source control system.
+	PRNumber string
 }
 
 // DetectVars detects and returns the CI variables for the current environment.
 func DetectVars() Vars {
-	v := Vars{Name: DetectSystem()}
-	// All CI systems have a name (that's how we know we're in a CI system). After detecting one, we will
-	// try to detect some additional CI-specific metadata that the CLI will use. It's okay if we can't,
-	// we'll just have reduced functionality for our various CI integrations.
+	if os.Getenv("PULUMI_DISABLE_CI_DETECTION") != "" {
+		return Vars{Name: ""}
+	}
+
+	systemName := DetectSystem()
+	// If CI variables have been set specifically for Pulumi in the environment,
+	// use that in preference to attempting to automatically detect the CI system.
+	// This allows Pulumi to work with any CI system with appropriate configuration,
+	// rather than requiring explicit support for each one.
+	if os.Getenv("PULUMI_CI_SYSTEM") != "" {
```

Comment:
This doesn't seem right. Wouldn't `DetectSystem()` return the value `PulumiCI` (or preferably `GenericCI`)? And in that case, we don't need to specifically set all of these values, rather `detectCISystemVars` would "just work" for that system?

Reply:
Hmm this block can be moved entirely. You are right that this shouldn't have to be a separate code block. I moved this from another function in order to keep the code to set Vars from build variables in one place.

I'll simply this function.